guides/source/action_view_helpers.md

```diff
@@ -412,29 +543,59 @@ Provides functionality for working with JavaScript in your views.
 
 #### escape_javascript
 
-Escape carrier returns and single and double quotes for JavaScript segments.
+Escapes carriage returns and single and double quotes for JavaScript segments.
+You would use this method to take a string of text and make sure that it doesn’t
+contain any invalid characters when the browser tries to parse it.
+
+For example, if you have a partial with a greeting that contains double quotes,
+you can escape the greeting to use in a JavaScript alert.
+
+**app/views/users/greeting.html.rb**
+
+```html+erb
+My name is <%= current_user.name %>, and I'm here to say "Welcome to our website!"
+```
+
+```html+erb
+<script>
+  var greeting = '<%= escape_javascript(render 'captured_pokemons/greeting') %>';
+  alert(`Hello, ${greeting}`);
+</script>
+```
+
+This will escape the double quotes correctly and display the greeting as in an
+alert box.
+
+See the [API
+Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/JavaScriptHelper.html#method-i-escape_javascript)
+for more information.
 
 #### javascript_tag
 
-Returns a JavaScript tag wrapping the provided code.
+Returns a JavaScript tag wrapping the provided code. You can pass a hash of
+options to control the behavior of the `<script>` tag.
 
 ```ruby
-javascript_tag "alert('All is good')"
+javascript_tag("alert('All is good')", type: 'application/javascript')
```

Comment:
Would it be useful to provide the alternative block version as an example as well?